pkg/controller/useraccount/useraccount_controller.go

```diff
@@ -240,6 +248,62 @@ func (r *ReconcileUserAccount) ensureIdentity(logger logr.Logger, userAcc *toolc
 	return identity, false, nil
 }
 
+func (r *ReconcileUserAccount) ensureNSTemplateSet(logger logr.Logger, userAcc *toolchainv1alpha1.UserAccount) (*toolchainv1alpha1.NSTemplateSet, bool, error) {
+	name := userAcc.Name
+
+	// create if not found
+	nsTmplSet := &toolchainv1alpha1.NSTemplateSet{}
+	if err := r.client.Get(context.TODO(), types.NamespacedName{Name: userAcc.Name, Namespace: userAcc.Namespace}, nsTmplSet); err != nil {
+		if errors.IsNotFound(err) {
+			logger.Info("creating a new NSTemplateSet", "name", name)
+			if err := r.setStatusProvisioning(userAcc); err != nil {
+				return nil, false, err
+			}
+			nsTmplSet = newNSTemplateSet(userAcc)
+			if err := controllerutil.SetControllerReference(userAcc, nsTmplSet, r.scheme); err != nil {
+				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
+					"failed to set controller reference for NSTemplateSet '%s'", name)
+			}
+			if err := r.client.Create(context.TODO(), nsTmplSet); err != nil {
+				return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err,
+					"failed to create NSTemplateSet '%s'", name)
+			}
+			logger.Info("NSTemplateSet created successfully", "name", name)
+			return nsTmplSet, true, nil
+		}
+		return nil, false, r.wrapErrorWithStatusUpdate(logger, userAcc, r.setStatusNSTemplateSetCreationFailed, err, "failed to get NSTemplateSet '%s'", name)
+	}
+	logger.Info("NSTemplateSet already exists", "name", name)
+
+	// update if not same
+	equal := nsTmplSet.Spec.CompareTo(userAcc.Spec.NSTemplateSet)
+	if !equal {
+		logger.Info("updating NSTemplateSet", "name", name)
+		if err := r.setStatusProvisioning(userAcc); err != nil {
```

Comment:
it shouldn't be set to `Provisioning` but `Updating`

Reply:
We don't have `Updating` reason.  Check other places too, https://github.com/codeready-toolchain/member-operator/blob/master/pkg/controller/useraccount/useraccount_controller.go#L178-L184

Reply:
But this logic triggers updates of the templates. If you don't want to incorporate update reason, then please move this code to the PR that will take care of updates. It will also make the PR smaller and easier to review

Reply:
Remove support for update, commit https://github.com/codeready-toolchain/member-operator/pull/53/commits/338c9bd783e3b050a223fb717a934db718a3df02